Automattic/sensei
# Fix auto-grading of manually graded questions

Fixes #1375 (may fix other quiz-related issues, as I found some other bugs that weren't related to the reported issue).

## Testing Instructions
1. Add a multiple choice and a True/False question to a lesson. Set their _Question grade_ to 1.
2. In _Quiz Settings_, ensure that _Grade quiz automatically_ is unchecked.
3. Sign up for the course and take the quiz. Answer the multiple choice question correctly and the True/False question incorrectly.
4. In WP admin, go to _Sensei LMS_ > _Grading_ and click the _Grade Quiz_ button for the relevant lesson and user.
5. Ensure that the questions haven't been graded yet (i.e. neither the checkmark nor the X icons should be highlighted, and the question shouldn't have a score).
6. Click the _Auto grade_ button.
7. Ensure both questions are graded properly.
8. Click the _Grade_ button to save.
9. Ensure the questions are still graded properly.

For bonus points, add more question types that can't be auto-graded (i.e. multi-line, single line, gap fill) to the lesson and ensure auto-grading only grades the multiple choice and True/False questions. (Note that you will need to remove the user from the course and start taking it again in order for new questions to show in the quiz).

Also, try checking the _Grade quiz automatically_ setting so that quizzes are auto-graded, and ensuring that only the multiple choice and True/False questions are automatically graded on submission. (Note that you will need to reset the learner's progress in order to take the quiz again).
